src/intrinsics/ecma262/Math.js

```diff
@@ -200,8 +201,18 @@ export default function(realm: Realm): ObjectValue {
 
   // ECMA262 20.2.2.27
   obj.defineNativeMethod("random", 0, context => {
-    if (realm.mathRandomGenerator !== undefined) {
-      return new NumberValue(realm, realm.mathRandomGenerator());
+    let mathRandomGenerator = realm.mathRandomGenerator;
+    if (mathRandomGenerator !== undefined) {
+      let loc = realm.currentLocation;
+      let error = new CompilerDiagnostic(
+        "Result of Math.random() is made deterministic via a fixed mathRandomSeed",
+        loc,
+        "PP0001",
```

Comment:
Please allocate a new error number in the range PP8000 - PP8999 and create a wiki page for it.